cmd/cmd.go

```diff
@@ -427,34 +442,35 @@ func RunGenerate(cmd *cobra.Command, args []string) error {
 		}
 
 		prompts = append([]string{string(in)}, prompts...)
+		opts.WordWrap = false
+		interactive = false
 	}
-
-	// output is being piped
-	if !term.IsTerminal(int(os.Stdout.Fd())) {
-		return generate(cmd, args[0], strings.Join(prompts, " "), false, format)
-	}
-
-	wordWrap := os.Getenv("TERM") == "xterm-256color"
+	opts.Prompt = strings.Join(prompts, " ")
 
 	nowrap, err := cmd.Flags().GetBool("nowordwrap")
 	if err != nil {
 		return err
 	}
-	if nowrap {
-		wordWrap = false
-	}
+	opts.WordWrap = !nowrap
 
-	// prompts are provided via stdin or args so don't enter interactive mode
-	if len(prompts) > 0 {
-		return generate(cmd, args[0], strings.Join(prompts, " "), wordWrap, format)
+	if !interactive {
```

Comment:
This seems like a potentially unrelated change in behavior? It seems like running `ollama run llama2 hello` will put me in interactive mode now?

Reply:
Good catch. Fixed now.